### PR TITLE
Fixes #23182 - Add param to limit upstream subs index to attachable ones

### PR DESCRIPTION
**Pull request by jturel**
Pretty sure we need to do this for the UI when we go to attach subs

### CODE REVIEW

File: app/models/katello/upstream_pool.rb [hunk 1]
```diff
@@ -4,10 +4,10 @@ class UpstreamPool < OpenStruct
 
     class << self
       def fetch_pools(params)
-        quantities_only = ::Foreman::Cast.to_bool(params.delete("quantities_only"))
+        quantities_only = ::Foreman::Cast.to_bool(params.delete(:quantities_only))
```
Comment (jturel) [inline app/models/katello/upstream_pool.rb]: I can change these back if there are objections but I do prefer to use symbols where we can

**Comment by jturel** [comment]
I think an argument could be made to make this the default behavior of the API since I don't think we have the use case of listing ALL upstream subs without filter, but with the param we are flexible.

**Comment by akofink** [comment]
> eliminating pools which don't have subs available or should otherwise be filtered out of the list

@jturel Could you please explain what exactly the `consumer` param does, since it's not documented in the Candlepin API docs? You say it removes pools that are already fully consumed. Anything else?

**Comment by jturel** [comment]
@akofink I don't know *exactly* what it does. Some other examples are: it removes subscriptions meant to be attachable to virtual guests only, and "dev" pools which I am also not clear on the  precise definition. In any case it's the correct way to filter available pools for a consumer. Candlepin folks could tell us more if we wanted to know all the details. Thanks for reviewing!

**Comment by akofink** [comment]
Ok, cool. I wonder if we should submit a CP bug for fixing up their docs? :stuck_out_tongue_closed_eyes:

**Comment by jturel** [comment]
Heh yeah, I see there are a few fields missing description on that particular API doc ...

**Pull request merged**